### PR TITLE
fix(experiments): Clean up warnings 

### LINKED ISSUE #11641: It's unclear a feature flag is still enabled when disabling an experiment
When disabling an experiment, the feature flag (currently) remains active. As a user, I'd expect disabling the experiment to also toggle off the feature flag.

## Options

1. Automatically disable and send all traffic to the winning variant (need user confirmation for this)
2. Automatically disable the feature flag and return to control (since the flag page says everything should be controlled through the experiment page)

    image

4. Easiest: Be more explicit the existing confirmation message (below) that the flag is still active until disabled

image

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Fixes a few more experiment bugs:

image

## Changes

Fixes https://github.com/PostHog/posthog/issues/11641 

image

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/experiments/experimentLogic.tsx [hunk 1]
```diff
@@ -332,8 +354,46 @@ export const experimentLogic = kea<experimentLogicType>({
         updateExperimentSuccess: async ({ experimentData }) => {
             actions.updateExperiments(experimentData)
         },
+        setNewExperimentData: async ({ experimentData }, breakpoint) => {
+            const experimentEntitiesChanged =
```
Comment (liyiy) [inline frontend/src/scenes/experiments/experimentLogic.tsx:358]: @neilkakkar you're the best but this is some python style code written in javascript instead LOL. It's not super clear to me what is happening here unless I thoroughly read through the entire function

Comment (liyiy) [reply frontend/src/scenes/experiments/experimentLogic.tsx:358]: It definitely works though just not the easiest to parse

**Review by liyiy — APPROVED** [review]

File: frontend/src/scenes/experiments/experimentLogic.tsx (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/scenes/experiments/experimentLogic.tsx:358]: Do you have any recommendations on how to make it clean? 😅

Comment (liyiy) [reply frontend/src/scenes/experiments/experimentLogic.tsx:358]: as examples I would bump `if (!experimentData.filters || Object.keys(experimentData.filters).length === 0) {` to the top, and probably just split up the display errors code under an action/listener that's named more explicitly, though that's probably what kea forms error handling will help with lol :P

**Pull request merged**